### ISSUE TITLE
Feature request: Enforce consistent formatting with remark-lint

### ISSUE THREAD

**Issue by RadValentin** [remoteintech/remote-jobs#366]
I'm making a suggestion to add [`remark-lint`](https://github.com/wooorm/remark-lint) to this repo. 

It should help ensure a consistent formatting across all your `*.md` files as well as helping with broken external links or mistyped internal ones (via [remark-lint-no-dead-urls](https://github.com/davidtheclark/remark-lint-no-dead-urls)). 

We could also run it on every PR through Travis, it's very easy to set that up.

Full list of supported rules [here](https://github.com/wooorm/remark-lint/blob/master/doc/rules.md). Let me know if I should open a PR.

**Comment by dougaitken** [comment]
That sounds interesting @RadValentin - something that comes up is that links can change _after_ they have been merged, will this tool look over the repository every so often and check?

**Comment by RadValentin** [comment]
`remark-lint` only checks for styling issues, it has plugins to check urls too and it displays a status in the console. It runs on node.

To run it periodically a [CI tool](https://en.wikipedia.org/wiki/Continuous_integration) is required. It can be used for all sorts of tasks.  
An usual scenario is that when someone opens a PR the CI tool kicks in. It sets up an instance, clones your repo, installs dependencies and runs your tests (unit, linting, etc.). It will then output a status on Github (pending/pass/fail).

<img width="766" alt="github-pr-tests-running" src="https://user-images.githubusercontent.com/3300066/33579053-1f1d8dcc-d951-11e7-9e7d-89ec776f8d15.png">

That was my original idea anyway and I proposed [Travis](https://travis-ci.org/) since it's a tool I'm familiar with. 

Let's just add the option to run `remark-lint` from the console for now, you can run it periodically to check url.  If that works OK then we can move on to setting up Travis in a separate issue.

**Comment by dougaitken** [comment]
If it's not too much trouble, then I'd be open to this to see if it does help. Might work if I finally get the list up on GH Pages

**Comment by RadValentin** [comment]
Seeing as how this repo might get merged [into another one](https://github.com/lukasz-madon/awesome-remote-job/issues/395#issuecomment-350036303), I'll wait for that to get resolved before starting work on this issue.

**Comment by dougaitken** [comment]
It won't be getting merged, if anything the two lists will sit side by side. Personally I see each list having a different angle on the same thing 👍 no rush about this anyway, I see it as a "nice to have" rather than needed.

**Comment by RadValentin** [comment]
@dougaitken I'm going to take a look at this later today, please feel free to assign this issue to me. Will open a PR tomorrow.
